content/api/commands/origin.md

```diff
@@ -0,0 +1,482 @@
+    cy.get('button#submit').click()
+  })
+
+  cy.origin('mycms.com', () => {
+    cy.url().should('contain', 'cms')
+    cy.get('#current-user').contains('logged in')
+    cy.get('button#edit-1').click()
+    cy.get('input#title').type('Updated title')
+    cy.get('button#submit').click()
+    cy.get('.toast').type('Changes saved!')
+  })
+
+  cy.visit('/items/1')
+  cy.get('h1').contains('Updated title')
+})
+```
+
+Always remember,
+[Cypress tests are not unit tests](https://docs.cypress.io/guides/references/best-practices#Creating-tiny-tests-with-a-single-assertion).
+
+### Serialization
+
+When entering a `cy.origin()` block, Cypress injects itself at runtime, with all
+your configuration settings, into the requested origin, and sets up
+bidirectional communication with that instance. This coordination model requires
+that any data sent from one instance to another be
+[serialized](https://developer.mozilla.org/en-US/docs/Glossary/Serialization)
+for transmission. It is very important to understand that variables **inside**
+the callback are not shared with the scope **outside** the callback. For example
+this will not work:
+
+**<Icon name="exclamation-triangle" color="red"></Icon> Incorrect Usage**
+
+```js
+const foo = 1
+cy.origin('somesite.com', () => {
+  cy.visit('/')
+  // This line will throw a ReferenceError because `foo` is not defined in the
+  // scope of the callback
+  cy.get('input').type(foo)
+})
+```
+
+Instead, the variable must be explicitly passed into the callback using the
+`args` option:
+
+**<Icon name="check-circle" color="green"></Icon> Correct Usage**
+
+```js
+const foo = 1
+cy.origin('somesite.com', { args: { foo } }, ({ foo }) => {
+  cy.visit('/')
+  // Now it will pass
+  cy.get('input').type(foo)
+})
+```
+
+Underneath the hood, Cypress uses
+[`JSON.stringify()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify)
+and
+[`JSON.parse()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse)
+(or equivalent library functions, implementation details may change) to
+serialize and deserialize the `args` object. This introduces a number of
+restrictions on the data which may be transmitted:
+
+- Primitive values only: object, array, string, number, boolean, null
+- Date objects will be converted to strings
+- Inherited properties are not serialized
+- Circular references are not allowed and will throw an error
+
+### Callback restrictions
+
+Because of the way in which the callback is transmitted and executed, there are
+certain limitations on what code may be run inside it. In particular, the
+following Cypress commands will throw errors if used in the callback:
+
+- `cy.origin()`
+- [`cy.intercept()`](/api/commands/intercept)
+- [`cy.session()`](/api/commands/session)
+- [`cy.server()`](/api/commands/server)
+- [`cy.route()`](/api/commands/route)
+- [`Cypress.Server.defaults()`](/api/cypress-api/cypress-server)
```

Comment:
This won't throw an error, we just don't sync big Cypress defaults between origins (e.g. Cypress.Server.defaults(), Cypress.Screenshot.defaults(), etc)

Reply:
@mschile I think we have an exception for [server](https://github.com/cypress-io/cypress/blob/feature-multidomain/packages/driver/src/multi-domain/unsupported_apis.ts#L11).

Reply:
Sorry, I'm a little confused here - I actually copied this list from the notes of one of the feature meetings & don't understand it entirely. Can you suggest a copy change?

Reply:
Sorry for the confusion. The error list you have is correct.

If needed, we can also state we don't sync Cypress defaults between origins.